doc/ext/traffic-server.py

```diff
@@ -371,6 +371,26 @@ def xref_cleanup(app, env, node, contnode):
                 return node
     return;
 
+def make_github_link(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """
+    This docutils role lets us link to source code via the handy :ts:git: markup.
+    Link references are rooted at the top level source directory. All links resolve
+    to GitHub.
+
+    Examples:
+
+        To link to proxy/Main.cc:
+
+            Hi, here is a link to the proxy entry point: :ts:git:`proxy/Main.cc`.
+
+        To link to CONTRIBUTING.md:
+
+            If you want to contribute, take a look at :ts:git:`CONTRIBUTING.md`.
+    """
+    url = 'https://github.com/apache/trafficserver/blob/master/{}'
```

Comment:
Not that I'm aware of.

I did some quick messing around with github URLs. It looks like this format allows you to view revisions of a file at specific commit ids:
`https://github.com/apache/trafficserver/blob/<commit_SHA>/<path_to_file>`.

Keeping track of which commit a reference is pinned to might be more tricky. Give me a few days to think about it.